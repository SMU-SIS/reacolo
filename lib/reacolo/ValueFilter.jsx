import React from 'react';
import ComponentFilter from './ComponentFilter';

const filter = (targetOrTargets, role) => {
  if (Array.isArray(targetOrTargets)) {
    return targetOrTargets.includes(role);
  }
  return targetOrTargets === role;
};

const ValueFilter = props => (
  <ComponentFilter rendered={filter(props.target, props.value)}>
    { props.children }
  </ComponentFilter>
);

/* eslint-disable react/forbid-prop-types */
ValueFilter.propTypes = {
  target: React.PropTypes.oneOfType([
    React.PropTypes.any.isRequired,
    React.PropTypes.arrayOf(React.PropTypes.any.isRequired)
  ]).isRequired,
  children: React.PropTypes.node.isRequired,
  value: React.PropTypes.any
};
/* eslint-enable react/forbid-prop-types */

ValueFilter.defaultProps = {
  value: undefined
};

export default ValueFilter;
