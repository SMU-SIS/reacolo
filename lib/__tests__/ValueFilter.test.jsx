/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import ValueFilter from '../reacolo/ValueFilter';

test('ValueFilter renders if the value is the target.', () => {
  const component = renderer.create(
    <ValueFilter target="test" value="test" >
      child
    </ValueFilter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ValueFilter does not render if the value role is not the target.', () => {
  const component = renderer.create(
    <ValueFilter target="test" value="something-else" >
      child
    </ValueFilter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ValueFilter renders if the value is one of the targets.', () => {
  const component = renderer.create(
    <ValueFilter target={['target1', 'target2', 'target3']} value="target2" >
      child
    </ValueFilter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ValueFilter does not render if the value role is not one of the targets.', () => {
  const component = renderer.create(
    <ValueFilter target={['target1', 'target2', 'target3']} value="something-else" >
      child
    </ValueFilter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
