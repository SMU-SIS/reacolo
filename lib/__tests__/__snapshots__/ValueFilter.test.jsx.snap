exports[`test ValueFilter does not render if the value role is not one of the targets. 1`] = `null`;

exports[`test ValueFilter does not render if the value role is not the target. 1`] = `null`;

exports[`test ValueFilter renders if the value is one of the targets. 1`] = `
<div>
  child
</div>
`;

exports[`test ValueFilter renders if the value is the target. 1`] = `
<div>
  child
</div>
`;
